File: frontend/lib/api/types/evidence.ts
export type EvidenceHubSkill = {
  id: string;
  name: string;
  category: string;
  extraction_method: string;
  evidence_confidence: number;
};

export type EvidenceHubSource = {
  label: string;
  document_name?: string | null;
  parsed_at?: string | null;
  repository_name?: string | null;
  repository_url?: string | null;
};

export type EvidenceHubItem = {
  id: string;
  source_type: string;
  source_reference: string | null;
  title: string | null;
  description: string | null;
  verification_status: string | null;
  strength: number | null;
  created_at: string;
  updated_at: string;
  skills: EvidenceHubSkill[];
  source: EvidenceHubSource;
};

export type EvidenceHubListResponse = {
  items: EvidenceHubItem[];
  total: number;
  limit: number;
  offset: number;
};

export type EvidenceHubQueryParams = {
  source_type?: string;
  skill?: string;
  search?: string;
  limit?: number;
  offset?: number;
};

File: backend/app/schemas/evidence.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, Field


class EvidenceHubSkillResponse(BaseModel):
    id: UUID
    name: str
    category: str
    extraction_method: str
    evidence_confidence: float


class EvidenceHubSourceResponse(BaseModel):
    label: str
    document_name: str | None = None
    parsed_at: datetime | None = None
    repository_name: str | None = None
    repository_url: str | None = None


class EvidenceHubItemResponse(BaseModel):
    id: UUID
    source_type: str
    source_reference: str | None
    title: str | None
    description: str | None
    verification_status: str | None
    strength: float | None
    created_at: datetime
    updated_at: datetime
    skills: list[EvidenceHubSkillResponse]
    source: EvidenceHubSourceResponse


class EvidenceHubListResponse(BaseModel):
    items: list[EvidenceHubItemResponse]
    total: int
    limit: int = Field(ge=1)
    offset: int = Field(ge=0)
